doc/design/couler_sqlflow.md

```diff
@@ -20,4 +20,80 @@ The migration includes the following parts:
 
 1. Instead of having multiple codegens, let us have only one, `codegen_couler.go`, which translates the [Intermediate Representation](/doc/design/intermediate_representation.md)(IR) of a SQL program into a Couler program. Then, SQLFlow can run the Couler compiler to convert further and execute the workflow.
 
-For example, `codegen_couler.go` converts a `SELECT ... TO TRAIN` statement into the call to `sqlflow.couler.{xgboost,tensorflow.elasticdl}.train(train_ir)`, which `train_ir` is the SQLFlow IR with JSON format.
+For example, `codegen_couler.go` converts a `SELECT ... TO TRAIN` statement into the call to `sqlflow.couler.{xgboost,tensorflow.elasticdl}.train(...)`.
+
+## Couler Step Function
+
+For example, an XGBoost train step can be like:
+
+``` python
+couler.run_container(
+    cmd='''
+IN_SQL="SELECT ... TO TRAIN xgboost.booster" sqlflow -parse -i $IN_SQL -o ir.proto_text &&
+python -m sqlflow_submitter.xgboost.train -ir ir.proto_text'''
+    env={"SQLFLOW_DATASOURCE": "mysql://user:pass@192.168.1.1:3306"} # set session message as the env vars.
+    image="sqlflow/sqlflow_submitter"
+)
+```
+
+From the above Couler function:
+- `sqlflow -parse` is **SQLFLow command-line tool**, which compiles the input extended SQL into an IR serialized file.
+- `sqlflow_submitter.xgboost.train` is **SQLFlow submitter Python module**, which accepts the IR protobuf text file and then submit a machine learning job.
+- `sqlflow/sqlflow_submitter` is a Docker image that packages the SQLFlow submitter Python module and SQLFlow command-line tool.
+
+### SQLFLow Command-line Tool
+
+`sqlflow -parse` is a command-line tool which accepts an extended SQL and outputs the SQLFlow IR with
+protobuf text format, the protobuf definition is as follows:
+
+```protobuf
+message FeatureColumn {
```

Comment:
`FeatureColumn` is different from `FieldMeta`. Should follow the struct in `intermediate_representation.go`.